plugins/httpapi/aci.py

```diff
@@ -0,0 +1,384 @@
+    def login(self, username, password):
+        """Log in to APIC"""
+        # Perform login request
+        self.connection.queue_message("step:", "Establishing connection to {0}".format(self.connection.get_option("host")))
+        method = "POST"
+        path = "/api/aaaLogin.json"
+        payload = {"aaaUser": {"attributes": {"name": username, "pwd": password}}}
+        data = json.dumps(payload)
+        try:
+            response, response_data = self.connection.send(path, data, method=method)
+            response_value = self._get_response_value(response_data)
+            self.connection._auth = {
+                "Cookie": "APIC-Cookie={0}".format(self._response_to_json(response_value).get("imdata")[0]["aaaLogin"]["attributes"]["token"])
+            }
+            self.connection.queue_message("step:", "Connection to {0} was successful".format(self.connection.get_option("host")))
+        except Exception as exc:
+            # Catch exception from HTTPError()
+            if "401" in str(exc):
+                raise
+
+    def logout(self):
+        method = "POST"
+        path = "/api/aaaLogout.json"
+        payload = {"aaaUser": {"attributes": {"name": self.connection.get_option("remote_user")}}}
+        data = json.dumps(payload)
+        try:
+            response, response_data = self.connection.send(path, data, method=method)
+        except Exception as exc_logout:
+            msg = "Error on attempt to logout from APIC. {0}".format(exc_logout)
+            raise ConnectionError(self._return_info(None, method, path, msg))
+        self.connection._auth = None
+        self._verify_response(response, method, path, response_data)
+
+    def set_parameters(self, method, path, data):
+        if self.params.get("port") is not None:
+            self.connection.set_option("port", self.params.get("port"))
+
+        if self.params.get("username") is not None:
+            self.connection.set_option("remote_user", self.params.get("username"))
+
+        if self.params.get("use_proxy") is not None:
+            self.connection.set_option("use_proxy", self.params.get("use_proxy"))
+
+        if self.params.get("use_ssl") is not None:
+            self.connection.set_option("use_ssl", self.params.get("use_ssl"))
+
+        if self.params.get("validate_certs") is not None:
+            self.connection.set_option("validate_certs", self.params.get("validate_certs"))
+
+        # The command timeout which is the response timeout from APIC
+        # If the persistent_connect_timeout is less than the response timeout from APIC the persistent socket connection will fail
+        if self.params.get("timeout") is not None:
+            self.connection.set_option("persistent_command_timeout", self.params.get("timeout"))
+            self.connection.set_option("persistent_connect_timeout", self.params.get("timeout") + 30)
```

Comment:
Why was the +30 decided for persistent_connect_timeout, and should a user be allowed to change this value?
Should we document this behaviour in the https://github.com/CiscoDevNet/ansible-aci/blob/29a664cf889fa0955bf66a45f972748948c57d09/plugins/doc_fragments/aci.py?